server/src/main/java/org/opensearch/action/support/TransportActions.java

```diff
@@ -64,4 +66,8 @@ public static boolean isReadOverrideException(Exception e) {
         return !isShardNotAvailableException(e);
     }
 
+    public static boolean inValidRetrySearchException(final Exception e) {
```

Comment:
Nitpick: Rename to `isNonRetryableSearchException`.

Alternatively, maybe `isRetryableSearchException` and invert the logic? (I'm not a big fan of "is not" boolean methods.)